### PR TITLE
FIX get_table_id should correctly handle quoted schema.tablename

### LINKED ISSUE #276: get_table_id does not correctly handle quoted schemas and tablenames
The `get_table_id` function throws a `ValueError` if the schema and tablename are quoted, which doesn't seem to be the correct behavior.

## Civis library version

1.9.2

## How to reproduce the bug

```python
In [1]: import civis

In [2]: client = civis.APIClient()

In [3]: client.get_table_id(table='foo.bar', database='baz')
Out[3]: 123

In [4]: client.get_table_id(table='"foo"."bar"', database='baz')
ValueError: Given table "foo"."bar" is not an exact match for returned table foo.bar.
```

## Desired behavior

```python
In [1]: import civis

In [2]: client = civis.APIClient()

In [3]: client.get_table_id(table='foo.bar', database='baz')
Out[3]: 123

In [4]: client.get_table_id(table='"foo"."bar"', database='baz')
Out[4]: 123
```

## Likely cause/fix

While `get_table_id` derives the schema and tablename via `civis.io.split_schema_tablename`:

https://github.com/civisanalytics/civis-python/blob/eb5d46c8ce14f1bb87b889b69c7fd9e1c751d777/civis/civis.py#L271

which (correctly) gives us `'foo'` as the schema and `'bar'` as the tablename (both without quotes) with `'"foo"."bar"'` (with quotes) as the input, it looks like the following bit (whose purpose is unclear to me) can be removed for the desired behavior:

https://github.com/civisanalytics/civis-python/blob/eb5d46c8ce14f1bb87b889b69c7fd9e1c751d777/civis/civis.py#L277-L280

That being said, perhaps there's more to it, and the current maintainers would probably know more.

### PR DESCRIPTION
**Pull request by jacksonllee**
Resolves #276

### CODE REVIEW

**Comment by jacksonllee** [comment]
All Travis builds have passed. Waiting for AppVeyor builds to finish: https://ci.appveyor.com/project/civisanalytics/civis-python/builds/22468450

**Comment by jacksonllee** [comment]
The AppVeyor builds have finished:

screen shot 2019-02-19 at 11 18 34 am

The Python 3.6 build failed, but the issue was the sort of non-deterministic test behavior we've seen from time to time that I think we can safely ignore for this PR:

```
================================== FAILURES ===================================
_____________________________ test_custom_scripts _____________________________
    def test_custom_scripts():
        with mock.patch.dict('os.environ', {'CIVIS_JOB_ID': '12',
                                            'CIVIS_RUN_ID': '40'}):
>           c = _check_executor(133)
civis\tests\test_futures.py:292: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
from_template_id = 133
    def _check_executor(from_template_id=None):
        job_id, run_id = 42, 43
        c = _setup_client_mock(job_id, run_id, n_failures=0)
        mock_run = c.scripts.post_containers_runs()
        if from_template_id:
            bpe = CustomScriptExecutor(from_template_id=from_template_id,
                                       client=c, polling_interval=0.01)
            future = bpe.submit(my_param='spam')
        else:
            bpe = _ContainerShellExecutor(client=c, polling_interval=0.01)
            future = bpe.submit("foo")
    
        # Mock and test running, future.job_id, and done()
        mock_run.state = "running"
>       assert future.running(), "future is incorrectly marked as not running"
E       AssertionError: future is incorrectly marked as not running
E       assert False
E        +  where False = <bound method Future.running of <ContainerFuture at 0x23e50ed518 state=succeeded returned Response>>()
E        +    where <bound method Future.running of <ContainerFuture at 0x23e50ed518 state=succeeded returned Response>> = <ContainerFuture at 0x23e50ed518 state=succeeded returned Response>.running
civis\tests\test_futures.py:257: AssertionError
```

This PR is ready for review.

File: civis/tests/test_civis.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    mock_client = mock.Mock()
+    mock_client.get_database_id.return_value = 123
+
+    mock_tables = mock.MagicMock()
+    mock_tables.__getitem__.side_effect = {0: mock.Mock()}.__getitem__
+
+    mock_client.tables.list.return_value = mock_tables
+
+    civis.APIClient.get_table_id(mock_client,
+                                 table=schema_tablename, database=123)
```
Comment (jacksonllee) [inline civis/tests/test_civis.py]: @stephen-hoover The Python 3 builds have passed, but the Python 2.7 one failed on this line (with `TypeError: unbound method get_table_id() must be called with APIClient instance as first argument (got Mock instance instead)`). Would it fine to just `@pytest.mark.skip` this test for Python 2.7? If not, I don't see how we can have a mock client object without also mocking out `get_table_id`. Or am I missing something?

Comment (stephen-hoover) [reply civis/tests/test_civis.py]: If you want a mock client object, it's better to use the `civis.tests.mocks.create_client_mock` function to get one (that will handle autospeccing for you). In this case, you might actually want to create a real client object and then replace the functions you want to mock with `mock.Mock` objects. What do you think of trying:
```python
from civis.tests.mocks import TEST_SPEC
client = APIClient(local_api_spec=TEST_SPEC, api_key='none')
client.get_database_id = mock.Mock(return_value=123)
```
and so forth. If you want to argue that there's so much mocking going on that the test isn't useful, I'm open to that. This test seems nice to have, but maybe a separate test on `civis.io.split_schema_tablename` is all that's needed here.

Comment (jacksonllee) [reply civis/tests/test_civis.py]: Ah, `TEST_SPEC` is what I didn't know about! The test has been updated, and the builds have passed.

**Comment by stephen-hoover** [comment]
It looks like there's still something going on with Travis. Why does it need to run tests on both your branch and the PR? Aren't those the same code? Maybe I'm misunderstanding what "Branch" and "Pull Request" mean there.

**Comment by jacksonllee** [comment]
I've contacted IT-Support to help sort out this repo's config (hopefully once and for all) after I'd noticed merging is still blocked. Will report back as soon as I have updates.

**Comment by jacksonllee** [comment]
According to IT-Support, apparently there are indeed both "branch" and "pull request" options for Travis CI checks, and somehow "branch" was checked -- it is not anymore, and hence everything green now.

**Pull request merged**